.github/workflows/publish.yaml

```diff
@@ -0,0 +1,29 @@
+name: Publish
+
+on:
+  push:
+    branches: [master]
+    tags: ['*']
+
+jobs:
+  publish-api-doc:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone the module
+        uses: actions/checkout@v2
+
+      - name: Setup Doxygen
+        run: sudo apt install -y doxygen
+
+      - name: Build module API documentation using Doxygen
+        run: |
+            cmake .
+            make module_api
+            sed -e "s%@PROJECT_.\+_DIR@/%%" Doxyfile.API.in > Doxyfile.API
+            doxygen Doxyfile.API
+
+      - name: Publish generated API documentation to GitHub Pages
+        uses: JamesIves/github-pages-deploy-action@4.1.4
+        with:
+          branch: gh-pages
+          folder: doc/api/html/
```

Comment:
Can we generate just one page from one header file using doxygen? Or copy `module_8h.html` to `index.html`?

I looked on the build documentation and it is a bit unexpected that one need to do several click to get the documention: open the page -> click `File list` -> click `module.h`.